issuance/issuance.go

```diff
@@ -344,11 +344,39 @@ type IssuerID int64
 
 // ID provides a stable ID for an issuer's certificate. This is used for
 // identifying which issuer issued a certificate in the certificateStatus table.
+// This value is computed as a truncated hash over the whole certificate,
+// meaning it is highly unique but not computable from end-entity certs.
 func (ic *Certificate) ID() IssuerID {
 	h := sha256.Sum256(ic.Raw)
 	return IssuerID(big.NewInt(0).SetBytes(h[:4]).Int64())
 }
 
+// IssuerNameID is a statistically-unique small ID which can be computed from
+// both CA and end-entity certs to link them together into a validation chain.
+// It is computed as a truncated hash over the issuer Subject Name bytes, or
+// over the end-entity's Issuer Name bytes, which are required to be equal.
+type IssuerNameID int64
+
+// NameID computes the IssuerNameID from an issuer certificate, i.e. it
+// computes a truncated hash over the issuer's Subject Name raw bytes. Useful
+// for storing as a lookup key in contexts that don't expect hash collisions.
+func (ic *Certificate) NameID() IssuerNameID {
+	h := crypto.SHA1.New()
```

Comment:
We should document why we use SHA1 (i.e., for compatibility with the widely deployed algorithm used in OCSP requests).

Reply:
Good call. Documented, and also extracted to ensure that the implementations of the two methods here never diverge.